utils/nctl/sh/assets/setup_shared.sh

```diff
@@ -432,45 +432,99 @@ function setup_asset_node_configs()
     done
 }
 
+function validators_string()
+{
+    local IDX
+    local COUNT_NODES_AT_GENESIS=$(get_count_of_genesis_nodes)
+    local PATH_TO_NET
+
+    local VALIDATORS
+
+    PATH_TO_NET="$(get_path_to_net)"
+
+    for IDX in $(seq 1 "$COUNT_NODES_AT_GENESIS")
+    do
+        VALIDATORS+="--validator $(cat "$PATH_TO_NET/nodes/node-$IDX/keys/public_key_hex"),$(get_node_pos_stake_weight "$COUNT_NODES_AT_GENESIS" "$IDX") "
+    done
+
+    echo $VALIDATORS
+}
+
 function setup_asset_global_state_toml() {
     log "... setting node global_state.toml"
 
     local COUNT_NODES=${1}
-    local PROTOCOL_VERSION=${2}
+    local TARGET_PROTOCOL_VERSION
     local IDX
     local GLOBAL_STATE_OUTPUT
-    local PATH_TO_NET
-    local PRE_1_4_0
-
-    PATH_TO_NET="$(get_path_to_net)"
+    local GLOBAL_STATE_VALIDATOR_OUTPUT
+    local PATH_TO_NET="$(get_path_to_net)"
+    local GLOBAL_STATE_UPDATE_TOOL_MINOR_VERSION
+    local STATE_ROOT_HASH
+    local PRE_14
+    local DATA_LMDB_PATH
+
+    # Check version of the global state update tool. This is not a proper semver check, for simplicity, we rely on the minor part only.
+    # 0.3 marks the transition to 1.5 fast sync node and supports the "validators" command.
+    GLOBAL_STATE_UPDATE_TOOL_MINOR_VERSION=$("$NCTL_CASPER_HOME"/target/"$NCTL_COMPILE_TARGET"/global-state-update-gen \
+        -h | grep "Global State Update Generator" | grep -oE '[^ ]+$' | cut -c3-3)
```

Comment:
nit: You can also use `global-state-update-gen --version` here. Also the `cut` only matches a single digit, so when we bump the minor version to something that has multiple digits it will not work.
We can use grep to match only the minor version like this:
`GLOBAL_STATE_UPDATE_TOOL_MINOR_VERSION=$("$NCTL_CASPER_HOME"/target/"$NCTL_COMPILE_TARGET"/global-state-update-gen --version | grep -oP '(?<=\d\.)\d+(?=\.\d+)'`

Reply:
See here: https://github.com/casper-network/casper-node/pull/3451#discussion_r1045730755